docs/tutorials/using-jsx-in-apostrophe.md

```diff
@@ -0,0 +1,673 @@
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 380px;
+  padding: 20px 10px;
+  margin: auto;
+  border-radius: 4px;
+  box-shadow: 0 3px 6px 0 #555;
+  background: white;
+  font-family: Montserrat;
+`;
+
+const AppLabel = styled.span`
+  color: black;
+  margin: 20px auto;
+  font-size: 18px;
+  font-weight: bold;
+`;
+
+function App({ defaultCity }) {
+  const [city, updateCity] = useState(defaultCity || "");
+  const [weather, updateWeather] = useState(null);
+
+  useEffect(() => {
+    if (defaultCity) {
+      fetchWeather(defaultCity);
+    }
+  }, [defaultCity]);
+
+  const fetchWeather = async (cityName) => {
+    try {
+      const response = await fetch(`/fetch-weather?city=${cityName}`);
+      const weather = await response.json();
+      updateWeather(weather);
+    } catch (error) {
+      console.error("Error fetching weather data:", error);
+    }
+  };
+
+  const handleFetchWeather = (e) => {
+    e.preventDefault();
+    fetchWeather(city);
+  };
+  return (
+    <Container>
+      <AppLabel>React Weather App</AppLabel>
+      <CityComponent updateCity={updateCity} fetchWeather={handleFetchWeather} />
+      {weather && <WeatherComponent weather={weather} city={city} />}
+    </Container>
+  );
+}
+
+export default App;
+```
+  <template v-slot:caption>
+    modules/react-weather-widget/ui/src/jsx-components/App.jsx
+  </template>
+
+</AposCodeBlock>
+
+It should be noted that the two components used by this React app are being imported in the `App.jsx` file that is imported in the base `ui/src/index.js` file. The Webpack build is clever enough to import all the files without having to import them to the base, as long as they are imported into a file that is imported into the base. The only other part of this code we need to focus on is the `fetchWeather()` function. In this app we have elected to use the [OpenWeatherMap](https://openweathermap.org/) API to retrieve the weather for each city. At the time of this writing it had a generous free tier, and easy geolocation from a city name. However, it does require an API key. We don't want to directly add this key into our `App.jsx` code since it will be exposed client-side. Instead, we are going to create a proxy endpoint in our project that will fetch the data and pass it back to our component.
+
+```javascript
+const response = await fetch(`/fetch-weather?city=${cityName}`);
+```
+This line in that function performs a fetch on the `/fetch-weather` endpoint, passing in the city name as a parameter.
+
+<AposCodeBlock>
+
+```javascript
+require('dotenv').config();
+
+module.exports = {
+  extend: '@apostrophecms/widget-type',
+  options: {
+    label: 'React Weather Widget'
+  },
+  fields: {
+    // schema field code
+  },
+  webpack: {
+    // webpack configuration code
+  },
+  apiRoutes(self) {
+    return {
+      get: {
+        '/fetch-weather': async function (req, res) {
```

Comment:
This isn't a public-facing, address bar URL, so I would do:

```
async fetchWeather(...) { ... }
```

And on the front end I'd access:

```
/api/v1/react-weather-widget/fetch-weather
```

This automatically namespaces the endpoint for you and allows you to use a more natural method name, it is automatically converted to kebab case.